### PR TITLE
Halloween scripts

### PR DESCRIPTION
**Pull request by MiladNazeri**
The following scripts are added in this PR:

1. scaryman.js : Tweening system for jump scares using models and timed sfx
2. scaryman_elevator.js : Special version to handle fbx that is animation based for the elevator
3. zone_trigger.js : System to create a standardized trigger for one or many entity uuids
4. emitter_3_with_loop.js : For handeling audio that needs to be looped
5. tween.js : Added a Module folder in Utilities and included this as a first module dependency
6. Polyfill.js : A few ES6 methods that are nice to have

All these scripts are in production currently, do they still need test plans?  Please let me know and I will include whatever is missing.

### CODE REVIEW

File: DomainContent/Halloween/Jump-Scares/scaryman.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+//  Created by Milad Nazeri on 09/11/18
+//  Copyright 2018 High Fidelity, Inc.
+//
+//  Tween based that is triggered from an enterEntity event
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+(function () {
+    /* eslint-disable indent */
```
[the remark this answers is not in the document]
Comment (MiladNazeri) [reply DomainContent/Halloween/Jump-Scares/scaryman.js]: The way I'm doing my code folding now, it messes with the editor.  I comment by section and this allows me to easily fold on those sections when you switch the editor mode to indention in vsCode

File: DomainContent/Halloween/Jump-Scares/scaryman_elevator.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+                        }, musicDelay);
+
+                        fxPosition = fxPosition ? fxPosition : currentPosition;
+                        Script.setTimeout(function() {
+                            playSound(fxObject, fxInjector, fxPosition, fxVolume, false, fxOffset);
+
+                        }, fxDelay);
+                        Entities.editEntity(_entityID,{ 
+                            animation: { loop: false, hold: false, running: true, currentFrame: 20, firstFrame: 20
```
[the remark this answers is not in the document]
Comment (MiladNazeri) [reply DomainContent/Halloween/Jump-Scares/scaryman_elevator.js]: done

File: DomainContent/Halloween/Jump-Scares/zone_trigger.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    var _userData = null;
+    var _userDataProperties = null;
+    var idsToFindKeys = [];
+    var MAX_COUNTER = 10; 
+    var TIME_OUT_INTERVAL = 500;
+    var SEARCH_RADIUS = 300;
+
+    this.preload = function(entityID) {
+        _entityID = entityID;
+        var position = Entities.getEntityProperties(entityID,'position').position;
```
[the remark this answers is not in the document]
Comment (MiladNazeri) [reply DomainContent/Halloween/Jump-Scares/zone_trigger.js]: done

File: Utilities/Polyfills.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    if (!Array.prototype.findIndex) {
+        Object.defineProperty(Array.prototype, 'findIndex', {
+            value: function (predicate) {
+                // 1. Let O be ? ToObject(this value).
+                if (this == null) {
+                    throw new TypeError('"this" is null or not defined');
+                }
+
+                var o = Object(this);
```
[the remark this answers is not in the document]
Comment (MiladNazeri) [reply Utilities/Polyfills.js]: These are actually all polyfills taken from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/

I updated the header though to reflect that

**Comment by misslivirose** [comment]
@AndySeattle given that this has been in Halloween a while, are we good to call this tested? :)

**Comment by AndySeattle** [comment]
Yes? Never received a QA Needed tag, but yes I am good with these being merged in.

**Pull request merged**